### PR TITLE
fix: Try-Catch typing

### PR DESCRIPTION
**Pull request by JonathanStarup**

### CODE REVIEW

**Comment by JonathanStarup** [comment]
@mlutze I hope this doesn't conflict with anything, but I guess it should be fixed in the generator?

**Comment by mlutze** [comment]
> @mlutze I hope this doesn't conflict with anything, but I guess it should be fixed in the generator?

It's likely also broken on my branch. But the test will catch it.

File: main/src/ca/uwaterloo/flix/language/phase/TypeInference.scala [hunk 1]
```diff
@@ -889,7 +889,10 @@ object TypeInference {
       case KindedAst.Expr.TryCatch(exp, rules, loc) =>
         val rulesType = rules map {
           case KindedAst.CatchRule(sym, clazz, body) =>
-            visitExp(body)
+            for {
+              _ <- unifyTypeM(sym.tvar, Type.mkNative(clazz, sym.loc), sym.loc)
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/TypeInference.scala:893]: What about being an instance of Throwable?

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/TypeInference.scala:893]: Does this also need to be fixed in ConstraintGeneration?

Comment (JonathanStarup) [reply main/src/ca/uwaterloo/flix/language/phase/TypeInference.scala:893]: > What about being an instance of Throwable?

I don't know how to do that. This PR still improves things though

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/TypeInference.scala:893]: CC @mlutze

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/TypeInference.scala:893]: I guess your test will help Matt make the required refactor.

Comment (JonathanStarup) [reply main/src/ca/uwaterloo/flix/language/phase/TypeInference.scala:893]: Ive corrected the constraint generator

**Pull request merged**